src/NJsonSchema/Generation/JsonSchemaGenerator.cs

```diff
@@ -351,9 +351,10 @@ public virtual void ApplyDataAnnotations(JsonSchema schema, JsonTypeDescription
             var contextualType = typeDescription.ContextualType;
 
             dynamic displayAttribute = contextualType.ContextAttributes.FirstAssignableToTypeNameOrDefault("System.ComponentModel.DataAnnotations.DisplayAttribute");
-            if (displayAttribute != null && displayAttribute.Name != null)
+            if (displayAttribute != null)
             {
-                schema.Title = displayAttribute.Name;
+                // GetName returns null if the Name property on the attribute is not specified.
+                schema.Title = displayAttribute.GetName();
```

Comment:
Is using GetName() a breaking change from what we had before?
Maybe we should add a null check here as well (as before) as title might already be set and gets resetted to null otherwise.

Reply:
GetName() will return the value of the Name field if the ResourceType property on the DisplayAttribute is null, so in this circumstance, the behaviour is the same.

If a ResourceType property is specified on the DisplayAttribute, then GetName() will retrieve the value from the ResourceType instead e.g.

```csharp
public class MyClass {
  // No ResourceType specified: DisplayAttribute.GetName() returns "Some_Property"
  [Display(Name = "Some_Property")]
  public string SomeProperty { get; set; }
}


public class MyOtherClass {
  // ResourceType specified: DisplayAttribute.GetName() returns MyResources.Some_Property
  [Display(ResourceType = typeof(MyResources), Name = "Some_Property")]
  public string SomeProperty { get; set; }
}
```

Current behaviour does not have a check to see if the title has already been set.

Reply:
Committed a check to ensure that the `Name` property is not null before calling GetName()

Reply:
Instead of a .Name check and know the internals of GetName() I think it would be better to do: 

```
if (displayAttribute != null)
{
     var name = displayAttribute.GetName();
     if (name != null)
     {
         schema.Title = name;
     }
}
```

this way we only use GetName() and not assume a correlation between Name and GetName() (same for description)

Reply:
Okay, will do.

Reply:
Done: https://github.com/RicoSuter/NJsonSchema/pull/1313/commits/3aa50c8ced3fb51c89f648991bbb623d7b94b907